pandas/io/tests/test_pickle.py

```diff
@@ -284,6 +284,46 @@ def test_pickle_v0_15_2(self):
         #
         tm.assert_categorical_equal(cat, pd.read_pickle(pickle_path))
 
+    def compression_explicit(self, compression):
+        # issue 11666
+        if compression == 'xz':
+            tm._skip_if_no_lzma()
+        with tm.ensure_clean(self.path) as path:
+            df = tm.makeDataFrame()
+            df.to_pickle(path, compression=compression)
+            df2 = pd.read_pickle(path, compression=compression)
```

Comment:
I like these roundtrip pickle tests, but I'm worried about a failure scenario where both `to_pickle` and `read_pickle` are broken in the same way. Therefore, I think it makes sense to also test reading of pickles which were compressed externally. I.e. use the command line to compress a pickled file, track in repo, and use these files to test `read_pickle`.